flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxExecutorImpl.java

```diff
@@ -68,14 +70,23 @@ public void executeFirst(
 
 	@Override
 	public void yield() throws InterruptedException {
-		mailbox.take(priority).run();
+		Mail mail = mailbox.take(priority);
+		try {
+			mail.run();
+		} catch (Exception ex) {
+			throw new FlinkRuntimeException(ex);
```

Comment:
Does wrapping into FlinkRuntimeException provide any additional info compared to RuntimeException?

Reply:
A little bit - it allows us/user to classify runtime exceptions that originate from Flink. Also this is I think our convention and potentially a bigger discussion whether we need `FlinkException`/`FlinkRuntimeException` hierarchy.

Reply:
We don't know here whether the exception originated from user-code or internal Flink code.

Reply:
That's right about the "cause" of this exception that we are throwing here. But this exception is being thrown by us.